File: go/rna-transcription/rna_transcription.go
package strand

import (
	//"strings"
)

var translate = map[rune]byte {
	'A':'U',
	'T':'A',
	'C':'G',
	'G':'C',
}

func ToRNA(dna string) string {

	rna := make([]byte,len(dna))
	for i,c := range dna {
		rna[i] = translate[c]
	}
	return string(rna)
	/*rna :=""
	for _,r := range dna {
		rna += string(translate[r])
	}
	return rna*/

	/*return strings.Map(func(r rune) rune {
		switch r {
		case 'A':
			return 'U'
		case 'T':
			return 'A'
		case 'C':
			return 'G'
		case 'G':
			return 'C'
		default:
			return r
		}
	},dna)*/
	// arn :=""
	// for _,c :=range dna {
	// 	if c =='A' {
	// 		c = 'U'
	// 		arn += string(c)
	// 	}else if c == 'T' {
	// 		c = 'A'
	// 		arn += string(c)
	// 	}else if c == 'C' {
	// 		c = 'G'
	// 		arn += string(c)
	// 	}else {
	// 		c = 'C'
	// 		arn += string(c)
	// 	}
	// }
	// return arn
}

File: typescript/rna-transcription/rna-transcription.ts
export function toRna(rna: string): string {

  for (let i = 0 ; i < rna.length;i++) {
    if (rna[i] === 'T') {
      rna = rna.slice(0, i) + 'A' + rna.slice(i + 1);
    }else if (rna[i] === 'A') {
      rna = rna.slice(0, i) + 'U' + rna.slice(i + 1);
    }else if (rna[i] === 'C') {
      rna = rna.slice(0, i) + 'G' + rna.slice(i + 1);
    }else if (rna[i] === 'G') {
      rna = rna.slice(0, i) + 'C' + rna.slice(i + 1);
    }else {
      throw new Error('Invalid input DNA.');
    }
  }
  return rna;
  
}
